src/openzaak/components/zaken/brondatum.py

```diff
@@ -187,7 +196,16 @@ def get_brondatum(
     elif (
         afleidingswijze == BrondatumArchiefprocedureAfleidingswijze.vervaldatum_besluit
     ):
-        # TODO: Relation from Zaak to Besluit is not implemented yet...
-        raise NotImplementedError
+        zaakbesluiten = zaak.besluit_set.all()
+        if not zaakbesluiten.exists():
+            # Cannot use vervaldatum_besluit if Zaak has no Besluiten
+            raise DetermineProcessEndDateException(
+                _("Geen besluiten aan zaak gekoppeld om brondatum uit af te leiden.")
+            )
+
+        max_vervaldatum = zaakbesluiten.aggregate(Max("vervaldatum"))[
```

Comment:
Can you add a test where there _does_ exist a `ZaakBesluit`, BUT the `vervaldatum` is `None`? That should return a validation error saying the vervaldatum must be set first or something like that?

@joeribekker sanity check that I'm not talking nonsense?

Reply:
It can be `None` so thats a valid case.